app/controllers/users_controller.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+# :nodoc:
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      sign_in @user
+      user @user
+      flash[:success] = 'You are signed up'
+      redirect_to '/login'
+    else
+      render :new
+    end
+  end
+
+  def show
+    @user = User.find(params[:id])
+    @attended_events = @user.attended_events
+    @events = @user.events
+  end
```

Comment:
I don't see you are showing the previous and upcoming events attended for an specific user. Also, check this error when I try to check one user: #8